node/src/node_configurator/mod.rs

```diff
@@ -131,13 +134,20 @@ pub fn data_directory_from_context(
                 .to_string();
             let right_local_data_dir =
                 wrong_local_data_dir.replace(&wrong_home_dir, &right_home_dir);
-            PathBuf::from(right_local_data_dir)
-                .join("MASQ")
-                .join(chain.rec().literal_identifier)
+            create_path_buf(chain, right_local_data_dir.to_string())
         }
     }
 }
 
+pub fn create_path_buf(
```

Comment:
The name is too vague. Have it say what actually is happening inside. The fact you need to use PathBuf isn't relevant; what would be helpful though is something such as: "add_masq_directories" or "add_chain_specific_directories" etc.

Reply:
changed